test/gnat_test.exs

```diff
@@ -8,6 +8,20 @@ defmodule GnatTest do
     :ok = Gnat.stop(pid)
   end
 
+  test "connect to a server with authentication" do
+    connection_settings = %{
+      host: 'localhost',
+      port: 4223,
+      tcp_opts: [:binary],
+      username: "bob",
+      password: "alice"
+    }
+    {:ok, pid} = Gnat.start_link(connection_settings)
+    assert Process.alive?(pid)
+    :ok = Gnat.ping(pid)
```

Comment:
oh, is this the reason you wanted expose `ping` as a public function? I guess it's a pretty good way to verify that the connection was successful...  It's certainly simpler than doing a `sub` + `pub` + `receive`